### PR TITLE
#99 - Add Strict Typing

### PR DESCRIPTION
**Pull request by chrom**

### CODE REVIEW

**Comment by naydav** [comment]
Missed classes:
\Magento\Inventory\Controller\Adminhtml\Source\Index
\Magento\Inventory\Controller\Adminhtml\Source\InlineEdit (also need to add type casting)
\Magento\Inventory\Controller\Adminhtml\Source\NewAction
\Magento\Inventory\Controller\Adminhtml\Source\Save
\Magento\Inventory\Controller\Adminhtml\Source\SourceCarrierDataProcessor
\Magento\Inventory\Controller\Adminhtml\Source\SourceHydrator
\Magento\Inventory\Controller\Adminhtml\Stock\Delete (also need to add type casting)
\Magento\Inventory\Controller\Adminhtml\Stock\Edit
\Magento\Inventory\Controller\Adminhtml\Stock\Index
\Magento\Inventory\Controller\Adminhtml\Stock\InlineEdit (also need to add type casting)
\Magento\Inventory\Controller\Adminhtml\Stock\MassDelete (also need to add type casting)
\Magento\Inventory\Controller\Adminhtml\Stock\NewAction
\Magento\Inventory\Controller\Adminhtml\Stock\Save
\Magento\Inventory\Controller\Adminhtml\Stock\StockSourceLinkProcessor (also need to add type casting)
Magento\Inventory\Indexer\* (all files in namespace)
Magento\Inventory\Model\OptionSource\* (all files in namespace)
Magento\Inventory\Observer\* (all files in namespace)
Magento\Inventory\Setup\* (all files in namespace)
Magento\Inventory\Test\* (all files in namespace)
Magento\InventoryApi\Test\* (all files in namespace)
Magento\InventoryCatalog\Model\* (all files in namespace)
Magento\InventoryCatalog\Plugin\Model\StockRepositoryPlugin
Magento\InventoryCatalog\Setup\InstallData
Magento\InventoryCatalog\Test\* (all files in namespace)
Magento\InventoryCatalog\Ui\Component\Control\Stock\DeleteButton
app\code\Magento\InventoryCatalog\registration.php (all registration files)

Also, it will be great if you will create static test fo check strict declaration (only for diff)

**Comment by chrom** [comment]
@naydav about test and all registration files think we need to discuss, if you do not mind.

File: dev/tests/static/testsuite/Magento/Test/Php/LiveCodeTest.php [hunk 1]
```diff
@@ -266,4 +271,22 @@ public function testCopyPaste()
             "PHP Copy/Paste Detector has found error(s):" . PHP_EOL . $output
         );
     }
+
+    public function testStrictTypes()
+    {
+        $filesMissingStrictTyping = [];
+
+        foreach (self::getWhitelist(['php'], '', '', '/_files/whitelist/strict_type.txt') as $fileName) {
+            $file = file_get_contents($fileName);
```
Comment (naydav) [inline dev/tests/static/testsuite/Magento/Test/Php/LiveCodeTest.php]: Is better to use token
http://php.net/manual/en/control-structures.declare.php

**Comment by maghamed** [comment]
#99

**Pull request merged**